### ISSUE TITLE
How to understand the empty normals saved?

### ISSUE THREAD

**Issue by SleepTight-Cat** [hbb1/2d-gaussian-splatting#210]
Hi authors,
I appreciate your work on 2D Gaussian Splatting—thank you for sharing it.
While reviewing the code, I came across a section in scene/gaussian_model.py that I found a bit unclear. Specifically, in the save_ply function:

`    def save_ply(self, path):
        mkdir_p(os.path.dirname(path))

        xyz = self._xyz.detach().cpu().numpy()
        normals = np.zeros_like(xyz)
        f_dc = self._features_dc.detach().transpose(1, 2).flatten(start_dim=1).contiguous().cpu().numpy()
        f_rest = self._features_rest.detach().transpose(1, 2).flatten(start_dim=1).contiguous().cpu().numpy()
        opacities = self._opacity.detach().cpu().numpy()
        scale = self._scaling.detach().cpu().numpy()
        rotation = self._rotation.detach().cpu().numpy()

        dtype_full = [(attribute, 'f4') for attribute in self.construct_list_of_attributes()]

        elements = np.empty(xyz.shape[0], dtype=dtype_full)
        attributes = np.concatenate((xyz, normals, f_dc, f_rest, opacities, scale, rotation), axis=1)
        elements[:] = list(map(tuple, attributes))
        el = PlyElement.describe(elements, 'vertex')
        PlyData([el]).write(path)`

I noticed that the normals array is initialized as all zeros (np.zeros_like(xyz)) before being written to the PLY file. Could you clarify the reasoning behind this? Is this a placeholder for future normal calculations (which I didn't find where), or is there another purpose for saving zeroed normals?
I would greatly appreciate any insights you can provide.
Thanks in advance for your time!

**Comment by hbb1** [comment]
Indeed it is a placeholder during the development. 

As you can see,  the third column of the rotation would be the normal by the 2D Gaussian's definition.  And now it should be able to export the ply with normal, say, oriented point cloud. The main problem here is that a 2D Gaussian assumes dual visibility so it may cause ambiguity when you load the ply into other softwares. 

In order to export a perfect oriented point cloud that can be viewed in blender or in meshlab, one should turn off the [DUAL_VISIBLE](https://github.com/hbb1/diff-surfel-rasterization/blob/main/cuda_rasterizer/auxiliary.h#L32) in the rasterizer and make sure to rebuild the rasterizer. Training with this rasterizer allows exporting oriented point cloud compatible with other visualizers.
